In [3]:
import pandas as pd

df_1=pd.read_csv('ex_crawlingdata.csv')
df_2=pd.read_csv('crawlingdata.csv',usecols=['date','id','content'])
df_2.columns=['Date','Id','Context']

print(len(df_1),len(df_2))

427 324


Length of df_1 and df_2 before this trial was 427 and 324

In [5]:
df_new1=df_1
df_new2=df_2
df=pd.concat([df_new1,df_new2])

In [6]:
name=["더비치크랩","맘앤타르트","이츠버거","제주프리또","지구상회"]
df["name"]=""
duplicate=df[(df["Context"].str.contains("더비치크랩"))|(df["Context"].str.contains("맘앤타르트"))|(df["Context"].str.contains("이츠버거"))|(df["Context"].str.contains("제주프리또"))|(df["Context"].str.contains("지구상회"))]

for x in name:
    duplicate.loc[duplicate["Context"].str.contains(x),"name"] = x

num=duplicate["name"].tolist()

C:\Users\suhao\AppData\Local\Temp\ipykernel_13272\3407827321.py:6: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  duplicate.loc[duplicate["Context"].str.contains(x),"name"] = x


In [7]:
ex=duplicate.index
extract=df.drop(ex)
#extract.to_csv("extract_name.csv",index=False)

Extract name by using gpt_test.py and open extract_name.csv again which have value in name column, and run under code

pip install emoji (to remove the emoji in text)

In [8]:
import pandas as pd
ex_df=pd.read_csv("extract_name.csv")
#change human after you update data
human=['하영면옥','똣똣라면','제주스타벅스,성산장인의집,산도롱맨도롱,우무','제주 더스푼','광양해장국','금오름','이츠버거','패스 더 솔트','네꼬야티하우스','보말칼국수','동문시장,홍가시나무,새별오름,금오름,몽상드애월','애월리에,스모키버거','우드나잇','라라블루','가람돌솥밥,자매국수,은희네해장국,그루브','해모둠','제주세원이네','난산리식당','사는게니나노','한림전복물회맛집','웨스트그라운드,런던베이글제주,레이어드제주,공천포식당,김희선몸국,연정식당','제주세원이네','친밀','마노르블랑,당근과 깻잎,선채향,노릇,올래국수','평대성게국수,영의네우럭명가,소라횟집','제주 자양식당','제주세원이네','제주세원이네','스시무사','제주세원이네','제주세원이네','제주세원이네','미풍해장국,한라축산정육식당,꽃담,양가형제','흑돼지바게트버거','제주세원이네','흑돼지박스성산점','제주세원이네','제주바솔트','이치찜제주시청점','화성식당,표선수산마트,오는정김밥,별난닭곰탕,가시아방국수,제갈양,교래흙돼지,논짓물','이츠버거','도주제김밥','봉자네숙성뼈갈비','노형농원','이츠버거','이치찜제주시청점','트리플몬스터버거','흑돼지박스성산점','제주세원이네','제주세원이네','흑돼지박스성산점','웃보','제주프리또','맹고','중문보말칼국수,써니 제주','달팽이식당','제주프리또','장수상회,보말이네,돈사돈','꾸아이띠아오팍,대포 주상절리','마농갈비','제주 프리또','괴델갈비구이','우동카덴,고등어봉초밥,서울앵무새제주,퀸아망,원앤온리제주','강정마을미나리,춘심이네,나인브릿지,마농치킨','제주해물밥','제주맛집칼국수','다정이네깁밤,우진해장국,육도담,해월정,고집돌우럭,히포파운드,노티드도넛,마마롱 카페','먹돌','호타루한담','톳짜장면','활어천국','소담한봉봉','장인의 집','봉기네깡통구이','은희네해장국','삿뽀로,만복반미','동귀리갈칫집,오는정김밥,돌미롱,우도땅콘,바다국수,도두해녀의집','도나망','중문수원','한라산볶음밥','평대바당국수','쇠소깍','전복리조또','우진해장국,산지내장탕','바닷속고등어쌈밥,돌카롱,명진전복,장인의집,명랑스낵,땅꼬,옹포리','어가,바릇수산,깡통흑돼지,이춘옥원조고등어쌈밥','오설록,상춘재,코기,돌카롱,성게라면,서광춘희,새우리깐새우김밥','어니스트밀크,김만복깁밥,부두횟집,당케 올레국수,델몬드 카페,덕인당보리빵','늘봄,곽지국수,덕성원,젠하이드어웨이제주,원앤온리제주','어진이네','순이네소금바치,보말칼국수,춘월식당','굳능상회','미영이네,돈사촌,우진해장국','소금바치 순이','제주희야국수,만선고등어회,알동네집화순점,중문정,추자숙이네밥상,말보르피노로제,말보로소비뇽블랑,게뷔르츠트라미너,몬탈풀치아노,아들레이드힐즈쉬라즈','오롯,난드르커피,자매국수','전복돌솥밥','형제도식당,평대리멍석','호돌이식당,산노루,다도,고사리식당,협재선술집','아토블리','맛타고라스라멘','제주앞바다횟집','버거요','까망돼지 서귀포점','탄탄흑돈양꼬치','제주산방산본점','애월브런치오,애월테라피,애월찜','제주도 신해바라기','템포 다이닝 라운지','템포 다이닝 라운지','뚱딴지 애월 본점','카페 치츠태비','탄탄흑돈양꼬치','제주정지','템포 다이닝 라운지','또시랑','탄탄흑돈양꼬치','카페 치즈태비','뽕끄레 주악','갈치바다,고이정,실비라면','제주도 번네식당','템포 다이닝 라운지','제주산방산본점','해저','해저','깡촌 흑돼지,빨간 흑돼지','템포 다이닝 라운지','템포 다이닝 라운지','템포 다이닝 라운지','fins cafe','깡촌 흑돼지,빨간 흑돼지','버스트 버거','템포 다이닝 라운지','템포 다우닝 라운지','제주정지','조천한식뷔페','템포 다이닝 라운지','해저','버스트 버거','템포 다이닝 라운지','템포 다이닝 라운지, 폴비앙 베이커리 카페,한우 오마카세','중문흑돼지','템포 다이닝 라운지, 폴비앙 베이커리 카페,한우 오마카세,50여개의 샴페인 페어링코스','돗담','연북로 해물탕 본점','중문흑돼지','템포 다이닝 라운지','사랑이꽃피는민박','돼지촌 제주본점','녹진녹진 두툼 딱새우','해저','템포 다이닝 라운지','해저','템포 다이닝 라운지','산방산 국수맛집','해저','템포 다이닝 라운지','순천미향제주산방산본점','템포 다이닝 라운지','템포 다이닝 라운지','템포 다이닝 라운지,폴비앙 베이커리 카페,한우 오마카세,50여개의 샴페인 페어링 코스','해물탕 본점']

In [11]:
import re
import emoji

name_lst=[]
for x in human:
    x=x.replace('#',',').replace('-',' ').replace('_',' ').replace('&',' ')
    a=x.split(',')
    for i,z in enumerate(a):
        z=emoji.demojize(z)
        z=re.sub(r"[^\w\s]","",z)
        clean_z=re.sub('[a-zA-Z]','',z).strip()
        a[i]=clean_z
    name_lst.extend(a)
name_lst=[k for k in name_lst if k]
name_lst.extend(num)

In [22]:
ex_name=pd.DataFrame({"name":name_lst})
jeju_name=ex_name.groupby("name",as_index=False).size().rename(columns={'size':'mention'})
Jeju_name=jeju_name.sort_values(by='mention',ascending=False)
Jeju_name

name  mention
42        더비치크랩      174
180        지구상회      129
171       제주프리또       44
72        맘앤타르트       19
191  템포 다이닝 라운지       19
..          ...      ...
78       몬탈풀치아노        1
79        몽상드애월        1
80         미영이네        1
81        미풍해장국        1
218       히포파운드        1

[219 rows x 2 columns]

In [24]:
Jeju_name["type"]=1
Jeju_name.to_csv("jeju_restaurant.csv",index=False)

Generate dataframe of Cafe

In [31]:
import pandas as pd
Cafe=pd.read_csv("crawlingdataCafe.csv",usecols=['date','id','content'])
Cafe.columns=['Date','Id','Context']
print(len(Cafe))
ex_cafe=Cafe[165:]
ex_cafe.to_csv("extract_cafe.csv",index=False)

246


In [ ]:
import pandas as pd

cafe=pd.read_csv("extract_cafe.csv",usecols=['Date','Id','name'])
#change human_cafe along you update data
human_cafe=['새빌','카페더라이트','지구상회','새빌','새빌','쿠리킨톤','웨이뷰','웨이뷰','마음에온 카페','웨이뷰','웨이뷰','웨이뷰','웨이뷰','카페더라이트','리그램 카페','카페더라이트','웨이뷰','웨이뷰','FIX THE FLAT,Kayaranooma,Shohwa Coffee,Toastory,Be Bleue,HEMU Cafe,Amore Full,Hanjalim Cafe,HEMU Delicafe,Daol Village Cafe,Js Table,Jeju Rurum,Seashell Cafe and Pottery,Grumpy Cat Cafe,Juanto Cafe,Cafe de Urban,The Anoyu,Monaco Cafe,The BOXO,Thera,Jakawhana Cafe,La Valse de la lune,Comma Coffee Jeju,Manjog Cafe,Dol Hareubang Cafe,The Seine Coffe','리그램,하라케케','카페더라이드','카페더라이트','어린왕자감귤밭','Wiee','지구상회','웨이뷰','새빌','쿠리킨톤','Cafe The Light','카페더라이트','리그램','새빌','브라보비치','새빌','새빌','서귀샌드','마음에온 카페','카페더라이트','Cafe Oreum,Cafe Kwangalli,Cafe Maram,Cafe Salang,Cafe Gohnganesang,Cafe Roan,Cafe Utopia,Cafe Forest,Cafe Eondudong,Cafe Gendael','브라보비치','카페더라이트','리그램','제주카페','새빌','위이,Wiee','벤처24,LOVE STREET ESPRESSO,야옹카페','카페더라이트','제주카페','새빌','어린왕자감귤밭','새빌','리그램','마음에온 카페','카페더라이트','새빌','카페더라이트','Cafe Delmar,Cafe Kala,Cafe Terarosa,Cafe Shab,Cafe Waiting,Cafe Altenburg','카페더라이트','카페더라이트','카페더라이트','Seolmundae Halmang Cafe,Street Coffee Stand,YourMoon Cafe,Cafe Delmoondo,Cafe Bomma,추억다방 Playground,Standby Cafe,Lee Joon-Seok Art Museum & Cafe,카페 앞마당,Monique Art Cafe,Cafe Firstmoon,Allee Caffe,Cafe Prodromica','새빌','제주카페','카페더라이트','새빌','새빌','세빌','Cafe Studio,Cafe Seol Hwa,Cafe Sulbing Jeju,Cafe Green Tear,Cafe Segu,Cafe De Aewol,Cafe Aloha,Cafe Delantti,Cafe Dezir,Cafe Oulette,Cafe Omu,Cafe Gabae,Cafe Tenten,Cafe Mini Chocolate,Cafe Artinaris,Cafe Greentable,Cafe Joseon Chocolate,Cafe Bok Joo,Cafe Yeoni,Cafe Siwon','새빌','새빌','D GLOBAL CAFE','리그램 제주카페','새빌','카페더라이트','안개산속 택배기사,제주새 치마바지그림방,니드보태,힐링집,메이크오늘,멍멍그림,시작과끝,솅지네로스트,솔의와이,버키와 사루마루','식물원카페,하이버드 제주,제주 카페,보라매바닐라,엔와하우스,하비터,딩크스커피,플라잉오렌지,에어콘카페,숨,공간,유빌,카킷,델리 미트 제주,제주스윙모카','카페더라이트','카페더라이트','꽃이만드는아이스크림,FULLMOON,늘푸름한 누룽지,시티가든,카페웨이브,해결사,코마코마','제주카페','지구상회,Earth Market Cafe,Cafe JoyFaily,Cafe Jaeny,Cafe curl e,Brisker Cafe,Cafe Cardinal,단다:빈Cafe,Cafe London London,하얼빈카페,Moby Dicks Cafe','새빌','제주카페,제주카페거리','카페더라이트','리그램','새빌','새빌','카페더라이트','제주카페,제주카페그라운드,제주카페마루,제주카페돛담,제주카페이야기,제주카페트리니티,제주카페마루','제주카페','제주카페','마음에온 카페','지구상회','카페더라이트','무야호 카페,망고힐리즈 카페,베이글 스토리 카페,와플오아시스 카페,매리치킨 카페,제사카 카페,맛나피자 카페,무지개 카페','티에르티에,엑스터시,쉐뚜,팔레드뭉클,아바,고응가,마찍,하이신성,올로아,비런,바다뚱,디자이너하우스,리논,제주도올레,자유로,탐폰카페,둥지카페','새빌','메종드레아리,카페폴리니아,티레바,카페레오,니뽕내뽕,카페티푸,카페마릴리아,수카라락,카페프론테라,카페롤링','제주카페','제주카페','제주카페','카페더라이트','Bomnal,Cafe Ferrero,오설록 제주 본점,Mediterranean Kitchen Jeju,Story-on,더 게하더 노을','제주카페','카페더라이트','리그램']

Length of cafe before this trial was 246

In [26]:
import re
import emoji

cafe_lst=[]
for y in human_cafe:
    y=y.replace('#',',').replace('-',' ').replace('_',' ').replace('&',' ')
    b=y.split(',')
    for j,s in enumerate(b):
        s=emoji.demojize(s)
        clean_s=re.sub(r"[^\w\s]","",s).strip()
        b[j]=clean_s
    cafe_lst.extend(b)
cafe_lst=[n for n in cafe_lst if n]
cafe_lst

['새빌',
 '카페더라이트',
 '지구상회',
 '새빌',
 '새빌',
 '쿠리킨톤',
 '웨이뷰',
 '웨이뷰',
 '마음에온 카페',
 '웨이뷰',
 '웨이뷰',
 '웨이뷰',
 '웨이뷰',
 '카페더라이트',
 '리그램 카페',
 '카페더라이트',
 '웨이뷰',
 '웨이뷰',
 'FIX THE FLAT',
 'Kayaranooma',
 'Shohwa Coffee',
 'Toastory',
 'Be Bleue',
 'HEMU Cafe',
 'Amore Full',
 'Hanjalim Cafe',
 'HEMU Delicafe',
 'Daol Village Cafe',
 'Js Table',
 'Jeju Rurum',
 'Seashell Cafe and Pottery',
 'Grumpy Cat Cafe',
 'Juanto Cafe',
 'Cafe de Urban',
 'The Anoyu',
 'Monaco Cafe',
 'The BOXO',
 'Thera',
 'Jakawhana Cafe',
 'La Valse de la lune',
 'Comma Coffee Jeju',
 'Manjog Cafe',
 'Dol Hareubang Cafe',
 'The Seine Coffe',
 '리그램',
 '하라케케',
 '카페더라이드',
 '카페더라이트',
 '어린왕자감귤밭',
 'Wiee',
 '지구상회',
 '웨이뷰',
 '새빌',
 '쿠리킨톤',
 'Cafe The Light',
 '카페더라이트',
 '리그램',
 '새빌',
 '브라보비치',
 '새빌',
 '새빌',
 '서귀샌드',
 '마음에온 카페',
 '카페더라이트',
 'Cafe Oreum',
 'Cafe Kwangalli',
 'Cafe Maram',
 'Cafe Salang',
 'Cafe Gohnganesang',
 'Cafe Roan',
 'Cafe Utopia',
 'Cafe Forest',
 'Cafe Eondudong',
 'Cafe Gendael',
 '브라보비치',
 '카페더라이트',
 '리그

In [27]:
cafe_name=pd.DataFrame({"name":cafe_lst})
jeju_cafe=cafe_name.groupby("name",as_index=False).size().rename(columns={'size':'mention'})
Jeju_cafe=jeju_cafe.sort_values(by='mention',ascending=False)
Jeju_cafe

name  mention
165          카페더라이트       22
123              새빌       21
153            제주카페       12
144             웨이뷰        9
102             리그램        6
..              ...      ...
64   Jakawhana Cafe        1
65       Jeju Rurum        1
66         Js Table        1
67      Juanto Cafe        1
186             힐링집        1

[187 rows x 2 columns]

In [28]:
Jeju_cafe.to_csv("jeju_cafe.csv",index=False)